### PR TITLE
Copy btn

### PR DESCRIPTION
**Pull request by jrchudy**
This PR addresses issue #790.

The changes include adding a `copy` button to the Record page that depends on the user being able to edit the record. This adds a url param `?copy=` and sets it to true. This url is constructed similar to our edit mode urls but uses this `copy` param to differentiate it and set `recordedit` in create mode.

You can test the chages [here](https://dev.isrd.isi.edu/~jchudy/chaise/record/#1/legacy:dataset/id=5). Assigning to myself and adding @howdyjessie as a reviewer.

### CODE REVIEW

File: record/record.controller.js [hunk 1]
```diff
@@ -33,6 +33,14 @@
             }
         };
 
+        vm.copyRecord = function() {
+            var referenceFilter = $rootScope.reference.location.filter;
```
Comment (howdyjessie) [inline record/record.controller.js]: Is `referenceFilter` being used anywhere else?/What is `referenceFilter` being used for?

File: recordedit/form.controller.js [hunk 2]
```diff
@@ -7,7 +7,7 @@
         var vm = this;
         var context = $rootScope.context;
         vm.recordEditModel = recordEditModel;
-        vm.editMode = context.filter || false;
+        vm.editMode = (context.filter && !context.copy) || false;
```
Comment (howdyjessie) [inline recordedit/form.controller.js]: The `|| false` is probably unnecessary here. `context.filter && !context.copy` will evaluate to `false` if it's not `true`.

Comment (howdyjessie) [inline recordedit/form.controller.js]: Or you may elect to keep the `|| false` if you feel it's more readable. I'm ambivalent.

File: record/record.controller.js (hunk 1 above)
Comment (jrchudy) [inline record/record.controller.js]: I accidentally forgot to remove this.

File: Makefile [hunk 3]
```diff
@@ -19,20 +19,27 @@ MODULES=node_modules
 BIN=$(MODULES)/.bin
 
 # Protractor scripts
+# Legacy apps tests
```
Comment (howdyjessie) [inline Makefile]: These comments though. 💯

**Pull request merged**